presto-main/src/main/java/io/prestosql/SystemSessionProperties.java

```diff
@@ -365,7 +376,9 @@ public SystemSessionProperties(
                         false,
                         value -> {
                             boolean spillEnabled = (Boolean) value;
-                            if (spillEnabled && featuresConfig.getSpillerSpillPaths().isEmpty()) {
+                            if (spillEnabled
+                                    && featuresConfig.getSpillerSpillPaths().isEmpty()
+                                    && (!serverConfig.isCoordinator() || nodeSchedulerConfig.isIncludeCoordinator())) {
```

Comment:
this is correct,
complex, 
weak (coordinator is unlikely included in any reasonable setup)
and insufficient (doesnt check whether workers have spill paths configured)

i am fine with this,
but i would also be fine with this check being removed altogether